### PR TITLE
rbd: add Sparsify implementing rbd_sparsify

### PR DESCRIPTION
**Pull request by phlogistonjohn**
The sparsify function can be used to make an image sparse by replacing
runs of zeros with holes. This change adds a wrapper for the basic
rbd_sparsify function and accompanying tests.

Fixes: #280

## Checklist
- [x] Added tests for features and functional changes
- [x] Public functions and types are documented
- [x] Standard formatting is applied to Go code

### CODE REVIEW

**Review by agarwal-mudit — APPROVED** [review]
LGTM
Small suggestion: Add a testcase where sparseSize > image size

File: rbd/rbd_nautilus.go [hunk 1]
```diff
@@ -101,3 +101,17 @@ func (image *Image) GetModifyTimestamp() (Timespec, error) {
 
 	return Timespec(ts.CStructToTimespec(ts.CTimespecPtr(&cts))), nil
 }
+
+// Sparsify makes an image sparse by deallocating runs of zeros.
+// The sparseSize value will be used to find runs of zeros and must be
+// a power of two no less than 4096 and no larger than the image size.
```
Comment (nixpanic) [inline rbd/rbd_nautilus.go:107]: @dillaman Can you comment on this description? I wonder if 4096 is enforced, or is that just the size of the RADOS objects/stripes or some other configurable value.

**Review by nixpanic — APPROVED** [review]
LGTM, but I would like Jason to confirm the 4096 fixed value.

File: rbd/rbd_nautilus.go (hunk 1 above)
Comment (dillaman) [reply rbd/rbd_nautilus.go:107]: https://github.com/ceph/ceph/blob/master/src/librbd/Operations.cc#L1675

**Pull request merged**